### PR TITLE
Add all changes

### PR DESCRIPTION
**Pull request by thcr1**
Create event

### CODE REVIEW

File: client/src/components/Captain.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Button, DialogActions, DialogContent } from "@mui/material";
+import Chatbot from "./Chatbot";
+import { BootstrapDialog, BootstrapDialogTitle } from "./helpers/SideDialog";
+
+
+export default function Captain() {
+
+  return (
+    <BootstrapDialog
+      // onClose={handleClose}
```
Comment (EdenItz) [inline client/src/components/Captain.tsx]: to be added?

File: client/src/components/Captain.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { Button, DialogActions, DialogContent } from "@mui/material";
+import Chatbot from "./Chatbot";
+import { BootstrapDialog, BootstrapDialogTitle } from "./helpers/SideDialog";
+
+
+export default function Captain() {
+
```
Comment (EdenItz) [inline client/src/components/Captain.tsx:51]: again?

File: client/src/components/Captain.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Button, DialogActions, DialogContent } from "@mui/material";
+import Chatbot from "./Chatbot";
+import { BootstrapDialog, BootstrapDialogTitle } from "./helpers/SideDialog";
+
+
```
Comment (EdenItz) [inline client/src/components/Captain.tsx:17]: leave room for jesus?

File: client/src/components/Captain.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+export default function Captain() {
+
+  return (
+    <BootstrapDialog
+      // onClose={handleClose}
+      aria-labelledby="customized-dialog-title"
```
Comment (EdenItz) [inline client/src/components/Captain.tsx]: do we need it?

File: client/src/components/Captain.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      open={true}
+    >
+      <BootstrapDialogTitle id="customized-dialog-title"
+        onClose={()=>undefined}
```
Comment (EdenItz) [inline client/src/components/Captain.tsx]: ha?

File: client/src/components/Captain.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        Captain ticket
+      </BootstrapDialogTitle>
+      <DialogContent dividers>
+        <Chatbot />
+      </DialogContent>
+      <DialogActions>
+        <Button autoFocus
+        // onClick={handleClose}
+        >
+          Save changes
```
Comment (EdenItz) [inline client/src/components/Captain.tsx]: maybe change the text to "exit chat" / "back to browsing"...

File: client/src/components/Chatbot.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import "@chatscope/chat-ui-kit-styles/dist/default/styles.min.css";
```
Comment (EdenItz) [inline client/src/components/Chatbot.tsx]: css import at the end of imports

File: client/src/components/Chatbot.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    MainContainer,
+    ChatContainer,
+    MessageList,
+    Message,
+    MessageInput,
+} from "@chatscope/chat-ui-kit-react";
+
+export default function Chatbot() {
+    return (
+        <div style={{ position: "relative", height: "500px" }}>
```
Comment (EdenItz) [inline client/src/components/Chatbot.tsx]: rem / rm instead of px

File: client/src/components/CreateEvent.tsx [hunk 9]
```diff
@@ -1,9 +1,121 @@
-import { Paper, Typography } from "@mui/material";
+import { Card, CardContent, Typography, TextField, Button, Grid, InputAdornment } from '@mui/material';
+import { InputEvent, Event } from "../graphql/graphql";
+import { useRef } from 'react';
```
Comment (EdenItz) [inline client/src/components/CreateEvent.tsx:1]: react imports at the top
cluster related imports together
css at the end of imports
node modules import at the start

File: client/src/components/CreateEvent.tsx [hunk 10]
```diff
@@ -1,9 +1,121 @@
...
+import { gql, useMutation } from "urql";
+import { useNavigate } from "react-router-dom";
+
```
Comment (EdenItz) [inline client/src/components/CreateEvent.tsx:3]: room for jesus

File: client/src/components/CreateEvent.tsx [hunk 11]
```diff
@@ -1,9 +1,121 @@
...
+    const inputEvent: InputEvent = {
+      name: eventNameRef.current?.value || '',
+      location: eventLocationRef.current?.value || '',
+      timeAndDate: eventTimeAndDateRef.current?.value || '',
+      type: eventTypeRef.current?.value || ''
+    };
+
+    // Call the createEvent mutation with the inputEvent object
+    createEvent({inputEvent}).then((result: any) => {
+
```
Comment (EdenItz) [inline client/src/components/CreateEvent.tsx]: space line

File: client/src/components/Events.tsx [hunk 12]
```diff
@@ -1,26 +1,26 @@
-import Grid from "@mui/material/Grid";
 import { useUserContext } from "../hooks/userController/userContext";
```
Comment (EdenItz) [inline client/src/components/Events.tsx:1]: imports order

File: client/src/components/Events.tsx [hunk 13]
```diff
@@ -1,26 +1,26 @@
...
 
-const ticketQuery = graphql(`
-  query ticketQuery {
-    ticket {
-      id
-      areaNumber
-    }
-  }
-`);
+// const ticketQuery = graphql(`
```
Comment (EdenItz) [inline client/src/components/Events.tsx:8]: nvm

File: client/src/components/helpers/SideDialog.tsx [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+export function BootstrapDialogTitle(props: DialogTitleProps) {
+  const { children, onClose, ...other } = props;
+
+  return (
+    <DialogTitle sx={{ m: 0, p: 2 }} {...other}>
+      {children}
+      {onClose ? (
+        <IconButton
+          aria-label="close"
```
Comment (EdenItz) [inline client/src/components/helpers/SideDialog.tsx]: do we need it?

File: client/src/components/helpers/SideDialog.tsx [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+          sx={{
+            position: 'absolute',
+            right: 8,
+            top: 8,
+            color: (theme) => theme.palette.grey[500],
+          }}
+        >
+          {/* <CloseIcon /> */}
+        </IconButton>
+      ) : null}
```
Comment (EdenItz) [inline client/src/components/helpers/SideDialog.tsx]: use && instead of : null

File: server/mongo/models/User.ts [hunk 16]
```diff
@@ -0,0 +1,34 @@
+import mongoose, { Schema, Types } from "mongoose";
+
```
Comment (EdenItz) [inline server/mongo/models/User.ts:2]: space?

File: server/src/server.ts [hunk 17]
```diff
@@ -1,19 +1,40 @@
...
+const cors = require('cors');
+const cookieParser = require('cookie-parser');
+
 import { createYoga } from "graphql-yoga";
 import getSchema from "./graphql/shcema";
 import settings from "./config/settings";
+import setUpMongo from "../mongo/mongoDbManager";
 
 (async () => {
+
```
Comment (EdenItz) [inline server/src/server.ts:11]: why all the space?

File: server/src/server.ts [hunk 18]
```diff
@@ -1,19 +1,40 @@
...
 
-  // Pass it into a server to hook into request handlers.
-  const server = createServer(yoga);
+  setUpMongo();
+
+  app.use(cookieParser());
+  
+  app.use(cors({
+    origin: [process.env.FRONTEND_ENDPOINT!, process.env.BACKEND_ENDPOINT!, "http://localhost:3000"],
+    methods: "DELETE, PUT, POST, GET, OPTIONS",
```
Comment (EdenItz) [inline server/src/server.ts:23]: no patche?

File: client/src/components/Captain.tsx (hunk 5 above)
Comment (thcr1) [reply client/src/components/Captain.tsx]: fix it later, I just want something to show and than I will improve it

File: server/mongo/models/User.ts (hunk 16 above)
Comment (thcr1) [reply server/mongo/models/User.ts:2]: not mine

File: server/src/server.ts (hunk 17 above)
Comment (thcr1) [reply server/src/server.ts:11]: not mine

File: server/src/server.ts (hunk 18 above)
Comment (thcr1) [reply server/src/server.ts:23]: not mine

File: client/src/components/Events.tsx (hunk 12 above)
Comment (thcr1) [reply client/src/components/Events.tsx:1]: not mine

File: client/src/components/Events.tsx (hunk 13 above)
Comment (thcr1) [reply client/src/components/Events.tsx:8]: I left it so you will be able to use it later

**Review by mayabetzalel — APPROVED** [review]

**Pull request merged**